src/TransponderCode.hpp

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+#include <cstdlib>
+#include <string>
+
+/**
+ * This class stores a transponder code.
+ * It accepts the code "as is" and converts it to an octal (base 8) 
+ * representation of the code. The output needs to be formatted with %04o to
+ * get a usable transponder code.
+ */
+class TransponderCode {
+  uint_least16_t value;
+
+public:
+  /**
+   * Uninitialized.
+   */
+  TransponderCode() = default;
+
+  /**
+   * Construct an empty instance.  Its IsDefined() method will return
+   * false.
+   */
+  static TransponderCode Null() noexcept {
+    return TransponderCode(9);
+  }
+
+  explicit TransponderCode(uint_least16_t code) noexcept {
+    SetCode(code);
+  }
+
+  explicit TransponderCode(const char* code) noexcept : value(ParseCode(code)) {}
+
+  void SetCode(uint_least16_t code) noexcept {
+    value = ParseCode(std::to_string(code).c_str());
+  }
+
+  void SetCode(const char* code) {
+    value = ParseCode(code);
+  }
+
+  /**
+   * Returns an octal (base 8) representaiton of the transponder code.
+   * To be formatted with %o to get human readable transponder code.
+   * 
+   * @return octal (base 8) unsigned
+   */
+  constexpr uint_least16_t GetCode() const noexcept {
+    assert(IsDefined());
+
+    return value;
+  }
+
+  constexpr bool IsDefined() const noexcept {
+    return value != 4096;
```

Comment:
What if the value is 4097? Or 65535? Why 4096?
Why not just use `code <= 07777` here?

Reply:
Done.